docs/tracing.md

```diff
@@ -0,0 +1,237 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(tracing-tutorial)=
+# Tracing
+
+To use `jax.jit` effectively, it is useful to understand how it works. Let's put
```

Comment:
Let's start with a sentence that introduces tracing a little more explicitly. I propose this but feel free to workshop:

`jax.jit` and other JAX transforms work by *tracing* a function to determine its effect on inputs of a specific shape and type. For a window into tracing, let's put a few print statements...: